pennylane/devices/default_tensor.py

```diff
@@ -238,22 +286,62 @@ def circuit(theta, phi, num_qubits):
             >>> circuit(theta, phi, num_qubits)
             [-0.9953099539219951, 0.0036631029671767208, 0.9999999876072984]
 
-            After the first execution, the time to run this circuit for 50 qubits is around 0.5 seconds depending on the machine.
+            After the first execution, the time to run this circuit for 50 qubits is around 0.5 seconds on a standard laptop.
             Increasing the number of qubits to 500 brings the execution time to approximately 15 seconds, and for 1000 qubits to around 50 seconds.
 
             The time complexity and the accuracy of the results also depend on the chosen keyword arguments for the device, such as the maximum bond dimension.
             The specific structure of the circuit significantly affects how the time complexity and accuracy of the simulation scale with these parameters.
+
+
+            We can also simulate quantum circuits using the Tensor Network (TN) method. This can be particularly useful for circuits that build up entanglement.
+            The following example shows how to execute a quantum circuit with the TN method and configurable depth using ``default.tensor``.
+
+            We set the contraction technique to ``"auto-split-gate"``. With this option, each gate is lazily added to the tensor network
+            and nothing is initially contracted, but the gate is automatically split if this results in a rank reduction.
+
+
+            .. code-block:: python
+
+                import pennylane as qml
+
+                phi = 0.1
+                dept = 10
+                num_qubits = 20
+
+                dev = qml.device("default.tensor", method="tn", contract="auto-split-gate")
+
+                @qml.qnode(dev)
+                def circuit(phi, depth, num_qubits):
+                    for qubit in range(num_qubits):
+                        qml.X(wires=qubit)
+                    for _ in range(depth):
+                        for qubit in range(num_qubits - 1):
+                            qml.CNOT(wires=[qubit, qubit + 1])
+                        for qubit in range(num_qubits):
+                            qml.RX(phi, wires=qubit)
+                    for qubit in range(num_qubits - 1):
+                        qml.CNOT(wires=[qubit, qubit + 1])
+                    return qml.expval(qml.Z(0))
+
+            >>> circuit(phi, dept, num_qubits)
```

Comment:
I am getting these two warnings when running even after installing pennylane and quimb. I guess these are optional in quimb?

```
[/Users/korbinianpalmarsson/anaconda3/envs/pennylane311/lib/python3.11/site-packages/cotengra/hyperoptimizers/hyper.py:33](https://file+.vscode-resource.vscode-cdn.net/Users/korbinianpalmarsson/anaconda3/envs/pennylane311/lib/python3.11/site-packages/cotengra/hyperoptimizers/hyper.py:33): UserWarning: Couldn't import `kahypar` - skipping from default hyper optimizer and using basic `labels` method instead.
  warnings.warn(
[/Users/korbinianpalmarsson/anaconda3/envs/pennylane311/lib/python3.11/site-packages/cotengra/hyperoptimizers/hyper.py:54](https://file+.vscode-resource.vscode-cdn.net/Users/korbinianpalmarsson/anaconda3/envs/pennylane311/lib/python3.11/site-packages/cotengra/hyperoptimizers/hyper.py:54): UserWarning: Couldn't find `optuna`, `baytune (btb)`, `chocolate`, `nevergrad` or `skopt` so will use completely random sampling in place of hyper-optimization.
  warnings.warn(
```

Maybe we could comment on them somewhere in the docs?

Reply:
Yep, this is quite annoying, and it is precisely the reason why I could not import this device in `devices/__init__.py`. This comes from `quimb` (or more precisely, from `cotengra`). I'm not sure what the best way to deal with it is at this point.